### PR TITLE
Fix querier limiter bug under multiselect

### PR DESCRIPTION
**Pull request by rajagopalanand**
**What this PR does**:

**Which issue(s) this PR fixes**:

This PR is a fix for https://github.com/cortexproject/cortex/issues/5620

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by yeya24 — COMMENTED** [review]
Let's add unit tests to make sure this uses the same limiter for multiple selects

File: pkg/querier/querier_test.go [hunk 1]
```diff
@@ -311,6 +311,201 @@ func TestShouldSortSeriesIfQueryingMultipleQueryables(t *testing.T) {
...
+		},
+		{
+			{Name: model.MetricNameLabel, Value: "bar"},
+			{Name: "orders", Value: "5"},
+		},
+	}
+
+	_, samples := mockTSDB(t, labelsSets, model.Time(start.Unix()*1000), int(chunks*samplesPerChunk), sampleRate, chunkOffset, int(samplesPerChunk))
+
+	unorderedResponse := client.QueryStreamResponse{
```
Comment (yeya24) [inline pkg/querier/querier_test.go:367]: Can we rename the var? No need to name it `unorderedResponse` here

**Comment by alanprot** [comment]
Nice! 

After we could also try to add a integ test on this.

File: pkg/querier/querier_test.go [hunk 2]
```diff
@@ -311,6 +311,201 @@ func TestShouldSortSeriesIfQueryingMultipleQueryables(t *testing.T) {
...
+		storeQueriables      []QueryableWithFilter
+		tenantLimit          validation.TenantLimits
+		query                string
+		assert               func(t *testing.T, r *promql.Result)
+	}{
+		{
+
+			name:                 "should result in limit failure for multi-select and an individual select hits the series limit",
+			description:          "query results in multi-select but duplicate finger prints get deduped but still results in # of series greater than limit",
+			query:                "sum_over_time(foo[1d:10s]) + avg_over_time(foo[1d:10s])",
```
Comment (yeya24) [inline pkg/querier/querier_test.go:426]: Can we simplify the query to be just `foo + foo`? Easier to understand

Comment (rajagopalanand) [reply pkg/querier/querier_test.go]: Done

File: pkg/querier/querier_test.go (hunk 1 above)
Comment (rajagopalanand) [reply pkg/querier/querier_test.go]: Done

**Comment by rajagopalanand** [comment]
> Nice!
> 
> After we could also try to add a integ test on this.



> Nice!
> 
> After we could also try to add a integ test on this.

Yes, will submit a separate PR for that

**Review by yeya24 — APPROVED** [review]
LGTM

**Pull request merged**